packages/js/client-config-builder/src/bundles/default.ts

```diff
@@ -0,0 +1,171 @@
+import { ClientConfigBuilder } from "../ClientConfigBuilder";
+import { BuilderConfig } from "../types";
+import * as ipfsHttpClient from "./embeds/ipfs-http-client/wrap";
+import * as ipfsResolver from "./embeds/async-ipfs-resolver/wrap";
+
+import { IWrapPackage, Uri } from "@polywrap/core-js";
+import { PluginPackage } from "@polywrap/plugin-js";
+import {
+  ethereumProviderPlugin,
+  Connection,
+  Connections,
+} from "@polywrap/ethereum-provider-js";
+import { httpPlugin } from "@polywrap/http-plugin-js";
+import { fileSystemPlugin } from "@polywrap/fs-plugin-js";
+import { loggerPlugin } from "@polywrap/logger-plugin-js";
+import { concurrentPromisePlugin } from "concurrent-plugin-js";
+import { ExtendableUriResolver } from "@polywrap/uri-resolver-extensions-js";
+import { IUriRedirect } from "@polywrap/core-js";
+
+// $start: getDefaultConfig
+export const ipfsProviders: string[] = [
+  "https://ipfs.wrappers.io",
+  "https://ipfs.io",
+];
+
+interface IDefaultEmbed {
+  uri: Uri;
+  package: IWrapPackage;
+  source: Uri;
+}
+
+interface IDefaultEmbeds {
+  ipfsHttpClient: IDefaultEmbed;
+  asyncIpfsResolver: IDefaultEmbed;
+}
+
+export const embeds: IDefaultEmbeds = {
+  ipfsHttpClient: {
+    uri: Uri.from("embed/ipfs-http-client@1.0.0"),
+    package: ipfsHttpClient.wasmPackage,
+    source: Uri.from("ens/wraps.eth:ipfs-http-client@1.0.0"),
+  },
+  asyncIpfsResolver: {
+    uri: Uri.from("embed/async-ipfs-uri-resolver-ext@1.0.0"),
+    package: ipfsResolver.wasmPackage,
+    source: Uri.from("ens/wraps.eth:async-ipfs-uri-resolver-ext@1.0.0"),
+  },
+};
+
+type UriResolverExtBootloader = [IDefaultEmbed, IUriRedirect, ...Uri[]];
+
+export const uriResolverExts: UriResolverExtBootloader = [
+  embeds.asyncIpfsResolver,
+  {
+    from: Uri.from("ens/wraps.eth:ens-text-record-uri-resolver-ext@1.0.0"),
+    to: Uri.from("ipfs/QmaM318ABUXDhc5eZGGbmDxkb2ZgnbLxigm5TyZcCsh1Kw"),
+  },
+  Uri.from("ens/wraps.eth:http-uri-resolver-ext@1.0.0"),
+  Uri.from("ens/wraps.eth:file-system-uri-resolver-ext@1.0.0"),
+  Uri.from("ens/wraps.eth:ens-uri-resolver-ext@1.0.0"),
+  Uri.from("ens/wraps.eth:ens-ipfs-contenthash-uri-resolver-ext@1.0.0"),
+  Uri.from("ens/wraps.eth:ens-ocr-contenthash-uri-resolver-ext@1.0.0"),
+];
+
+interface IDefaultPlugin {
+  uri: Uri;
+  plugin: PluginPackage<unknown>;
+  implements: Uri[];
+}
+
+interface IDefaultPlugins {
+  logger: IDefaultPlugin;
+  http: IDefaultPlugin;
+  fileSystem: IDefaultPlugin;
+  concurrent: IDefaultPlugin;
+  ethereumProvider: IDefaultPlugin;
+}
+
+export const plugins: IDefaultPlugins = {
+  logger: {
+    uri: Uri.from("plugin/logger@1.0.0"),
+    plugin: (loggerPlugin({}) as unknown) as PluginPackage<unknown>,
```

Comment:
Minor nit: I don't think we should do this casting here; as it erases the factory function's return type to `unknown` and `unknown` can be then cast to anything (it's as good as using `any`). Since `IDefaultPlugin` already states the `plugin` property is of type `PluginPackage<unknown>`, I think doing just `plugin: loggerPlugin({})` should automatically coerce the result of the factory function correctly to `PluginPackage<unknown>`. Same goes for the other plugins below in this structure.

Another thought is that if the generic in `PluginPackage` disallows polymorphic plugin lists (as we need to deterministically know which plugin package to properly pass the correct generic) and we need to frequently cast the generic as unknown; we may wanna review that interface definition.

Reply:
Yes I agree @namesty, should have added a `// TODO: ` to these casts, we need to add `@polywrap/core-js` as a `peerDependency` within the plugin wrappers, to help with package -> type resolution in typescript. This will be removed.

Reply:
Hey @namesty this is fixed here, had to update all plugins to properly use `peerDependency`s: https://github.com/polywrap/toolchain/pull/1599

Helpful reading: https://nodejs.org/es/blog/npm/peer-dependencies/